libvcx/src/disclosed_proof.rs

```diff
@@ -1024,33 +1036,40 @@ mod tests {
     #[cfg(feature = "general_test")]
     fn test_create_proof() {
         let _setup = SetupAriesMocks::init();
+        settings::set_config_value(settings::CONFIG_PROTOCOL_TYPE, "4.0");
 
-        assert!(create_proof("1", ::utils::constants::PROOF_REQUEST_JSON).unwrap() > 0);
+        assert!(create_proof("1", ARIES_PROOF_REQUEST_PRESENTATION).unwrap() > 0);
     }
 
     #[test]
     #[cfg(feature = "general_test")]
     fn test_create_fails() {
         let _setup = SetupAriesMocks::init();
+        settings::set_config_value(settings::CONFIG_PROTOCOL_TYPE, "4.0");
 
         assert_eq!(create_proof("1", "{}").unwrap_err().kind(), VcxErrorKind::InvalidJson);
     }
 
     #[test]
     #[cfg(feature = "general_test")]
-    #[cfg(feature = "to_restore")]
     fn test_proof_cycle() {
         let _setup = SetupAriesMocks::init();
+        settings::set_config_value(settings::CONFIG_PROTOCOL_TYPE, "4.0");
 
-        let connection_h = connection::tests::build_test_connection_inviter_invited();
+        let connection_h = connection::tests::build_test_connection_inviter_requested();
+
+        AgencyMockDecrypted::set_next_decrypted_response(GET_MESSAGES_DECRYPTED_RESPONSE);
+        AgencyMockDecrypted::set_next_decrypted_message(ARIES_PROOF_REQUEST_PRESENTATION);
 
         let request = _get_proof_request_messages(connection_h);
 
-        let handle = create_proof("TEST_CREDENTIAL", &request).unwrap();
-        assert_eq!(VcxStateType::VcxStateRequestReceived as u32, get_state(handle).unwrap());
+        let handle_proof = create_proof("TEST_CREDENTIAL", &request).unwrap();
+        assert_eq!(VcxStateType::VcxStateRequestReceived as u32, get_state(handle_proof).unwrap());
 
-        send_proof(handle, connection_h).unwrap();
-        assert_eq!(VcxStateType::VcxStateAccepted as u32, get_state(handle).unwrap());
+        // todo: mock dependencies to make generate_proof work
+        // generate_proof(proof_handle, selected_credentials.into(), "{}".to_string()).unwrap();
+        // send_proof(handle_proof, connection_h).unwrap();
+        // assert_eq!(VcxStateType::VcxStateAccepted as u32, get_state(handle_proof).unwrap());
```

Comment:
Should we leave this test here? We removed another test requiring credentials...